src/visitable.cpp

```diff
@@ -297,8 +297,8 @@ static VisitResponse visit_internal( const std::function<VisitResponse( item *,
             return VisitResponse::ABORT;
 
         case VisitResponse::NEXT:
-            if( node->is_gun() || node->is_magazine() ) {
-                // content of guns and magazines are accessible only via their specific accessors
+            if( node->is_tool() || node->is_gun() || node->is_magazine() ) {
```

Comment:
None as of yet but it's a bug waiting to happen. The only contents for a tool should be a `magazine` which is definitely not available for crafting etc